### PR TITLE
Drop MARK_REGION macro and update kernel label style

### PR DESCRIPTION
**Pull request by dalg24**
Related to #362 

Removing `MARK_REGION ` and `<ArborX_Macros.hpp>` header altogether since empty
Changing `ArborX_` to `ArborX::FOO::`

I tried to be consistent.  Please suggest something else if you don't like something.

### CODE REVIEW

**Comment by dalg24** [comment]
I have no objection to squashing the first 5 commits if you want me to.

File: src/details/ArborX_DetailsTreeConstruction.hpp [hunk 1]
```diff
@@ -69,7 +68,7 @@ inline void calculateBoundingBoxOfTheScene(ExecutionSpace const &space,
   using Access = AccessTraits<Primitives, PrimitivesTag>;
   auto const n = Access::size(primitives);
   Kokkos::parallel_reduce(
-      ARBORX_MARK_REGION("calculate_bounding_box_of_the_scene"),
+      "ArborX::TreeConstruction::calculate_bounding_box_of_the_scene",
```
Comment (aprokop) [inline src/details/ArborX_DetailsTreeConstruction.hpp:71]: Maybe
```suggestion
      "ArborX::BVH::TreeConstruction::calculate_bounding_box_of_the_scene",
```
but don't feel strong about it.

File: src/details/ArborX_DetailsTreeTraversal.hpp [hunk 2]
```diff
@@ -54,14 +53,14 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
     else if (bvh_.size() == 1)
     {
       Kokkos::parallel_for(
-          ARBORX_MARK_REGION("BVH:spatial_queries_degenerated_one_leaf_tree"),
+          "ArborX::TreeTraversal::spatial_queries_degenerated_one_leaf_tree",
```
Comment (aprokop) [inline src/details/ArborX_DetailsTreeTraversal.hpp:56]: Thought about suggesting adding `BVH` here, but then, tree traversal is pretty universal and should work with other trees, so maybe not.

File: test/tstKokkosToolsAnnotations.cpp [hunk 3]
```diff
@@ -55,6 +55,7 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(bvh_bvh_allocations_prefixed, Tree, TreeTypes)
          void const * /*ptr*/, uint64_t /*size*/) {
         std::cout << label << '\n';
         BOOST_TEST(
+            (isPrefixedWith(label, "ArborX::BVH::") || // data member
```
Comment (aprokop) [inline test/tstKokkosToolsAnnotations.cpp:58]: Thanks! :ashamed:

**Review by aprokop — APPROVED** [review]
I have few suggestions, but lets not think hard on any of this, and feel free to ignore all of them, it already significantly improves the status quo.

**Comment by aprokop** [comment]
> I have no objection to squashing the first 5 commits if you want me to.

Could be nice, but don't really care.

File: src/details/ArborX_DetailsTreeConstruction.hpp (hunk 1 above)
Comment (dalg24) [reply src/details/ArborX_DetailsTreeConstruction.hpp:71]: Well this very kernel could technically be used by the brute force impl

File: src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp [hunk 4]
```diff
@@ -279,14 +280,14 @@ queryDispatch(NearestPredicateTag, BVH const &bvh, ExecutionSpace const &space,
 {
   using MemorySpace = typename BVH::memory_space;
   Kokkos::View<Kokkos::pair<int, float> *, MemorySpace> out(
-      "pairs_index_distance", 0);
+      "ArborX::BVH::query::pairs_index_distance", 0);
```
Comment (dalg24) [inline src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp:283]: I forgot to highlight that change.  That was a label I forgot to edit.

File: test/tstKokkosToolsAnnotations.cpp (hunk 3 above)
Comment (dalg24) [reply test/tstKokkosToolsAnnotations.cpp:58]: I messed up on the next line forgot to remove opening (

**Pull request merged**